.github/workflows/builder_nodejs_slsa3.yml

```diff
@@ -91,6 +107,42 @@ jobs:
       # TODO(#888): Publish to GitHub release.
 
       # TODO(#1669): Publish to npm. Note that we need to upload our own bundle to npm.
-      # - name: Publish to npm registry
-      #   run: |
-      #     npm publish # ...
+      # TODO(#1669): Use mainline version of npm
+      - name: Clone npm fork
+        id: clone-fork
+        run: |
+          set -euo pipefail
+
+          # Clone npm fork.
+          cd "$(dirname "$(which node)")"
+          mkdir -p node_modules
+          cd node_modules
+          git clone https://github.com/ianlewis/cli.git npm
+
+          # checkout branch
+          cd npm
+          git checkout oob-provenance
+
+          # Install dependencies
+          ./bin/npm ci
+
+      # TODO(#1669): Use mainline version of npm
+      # TODO(#1669): Generalize directory path checking. The same logic is needed in internal action.
+      - name: Publish the package
+        id: publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.node-auth-token }}
+          ATTESTATION_DIR: ${{ steps.attestation-download.outputs.download-path }}
+          ATTESTATION_NAME: ${{ fromJSON(needs.slsa-run.outputs.build-artifacts-outputs).attestation-name }}
+          PACKAGE_DIR: ${{ steps.package-download.outputs.download-path }}
+          PACKAGE_FILE_NAME: ${{ fromJSON(needs.slsa-run.outputs.build-artifacts-outputs).package-filename }}
```

Comment:
I think all these env variables are untrusted (either provided by the TRW or the PW), so pre-pending `UNTRUSTED_` to all of us would be preferable?